core-to-plc/src/Language/Plutus/CoreToPLC.hs

```diff
@@ -90,19 +92,14 @@ sdToTxt sd = do
   (flags, _, _, _) <- ask
   pure $ T.pack $ GHC.showSDoc flags sd
 
-conversionFail :: (MonadError (Error ()) m, MonadReader ConvertingContext m) => GHC.SDoc -> m a
-conversionFail = (throwError . ConversionError) <=< sdToTxt
-
-unsupported :: (MonadError (Error ()) m, MonadReader ConvertingContext m) => GHC.SDoc -> m a
-unsupported = (throwError . UnsupportedError) <=< sdToTxt
+conversionFail :: (MonadError ConvError m, MonadReader ConvertingContext m) => GHC.SDoc -> m a
+conversionFail = (throwError . NoContext . ConversionError) <=< sdToTxt
```

Comment:
Why the parens? `(.)` binds very tightly, I haven't seen a case where it's unsafe to omit parens around `(.)`.

Reply:
I find it quite non-obvious to remember how the various function-composition-like operators associate, so I prefer to bracket it for my sanity.